irods/test/test_paths.py

```diff
@@ -0,0 +1,26 @@
+import unittest
+import os.path
+from irods.path import iRODSPath
+import re
+
+def _standard_path_normalizer (p, sep='/'):
+    leading_seps_deduplicator = re.compile("^{}+".format(re.escape(sep)))
+    npath = os.path.normpath(p)
+    if os.path.sep != sep:
+        return npath.sub(os.path.sep, sep)
+    return leading_seps_deduplicator.sub(sep,npath) # de-dupe leading separators - https://bugs.python.org/issue26329
+
+_normalization_test_paths = [
+    "/zone",
+    "/zone/home1/../home2",
+    "/zone/home1/user/./../trash",
+    "//zone//abc/",
+    "//zone//abc/.",
+]
+
+class MyTest(unittest.TestCase):
+    def testit(self):
+        for test_path in _normalization_test_paths:
+            norm_path = iRODSPath(test_path)
+            reference = _standard_path_normalizer(test_path)
```

Comment:
I know we test primarily on non-Windows systems at this point, but this test reinforces that. Would it be better to present more explicit "expected" results (aka `reference`) against which we can compare? Maybe the list of test paths can include their manually generated, normalized paths? In this way, we can avoid using `os.path.normath` here as well. Thoughts?

Reply:
I think it's a problem to use `os.path.normpath` in the test in the way it's being used, as its behavior is what we're trying to emulate?  But if it's stylistically better to provide the explicit expected results alongside the test cases, I'm willing to do that.  Maybe it would result in less "huh?" factor for those reading the test.  I'm fine either way, but inclined to leave things this way. @trel?

Reply:
Yep, just a suggestion for possible improvement. It's fine as-is as well